### ISSUE TITLE
Get "Missing required parameter 'symbol'" when calling spot submit order 

### ISSUE THREAD

**Issue by jordant97** [tiagosiebler/bybit-api#125]
My code as below, I am pretty sure I have all the required params in place. 
`
 let newOrder = await client.submitOrder({
      symbol: "BTCUSDT",
      qty: 1000,
      side: "Buy",
      type: "LIMIT",
      timeInForce: "IOC",
      price: 2.22,
    });
`

The response return from bybit api
`{
  ret_code: -1004,
  ret_msg: "Missing required parameter 'symbol'",
  ext_code: null,
  ext_info: null,
  result: null
}`

**Comment by reinierkors** [comment]
Your code looks fine. It's an issue with this package.

This is because Bybit does not support JSON. It [only supports](https://bybit-exchange.github.io/docs/spot/?console#t-constructingtherequest) ```application/x-www-form-urlencoded```.

**Comment by tiagosiebler** [comment]
Looks like the format is indeed different for POST requests in spot markets (compared to the older markets such as perps & futures). We will need to implement a separate request mechanism for spot markets.

**Comment by pranaypratyush** [comment]
I am getting the following for any private data endpoint on spot client. Is this because of this issue or separate? 
```
{
  ret_code: -1022,
  ret_msg: 'Signature for this request is not valid.',
  ext_code: null,
  ext_info: null,
  result: null
}
```

**Comment by GorWebfoundation** [comment]
> Looks like the format is indeed different for POST requests in spot markets (compared to the older markets such as perps & futures). We will need to implement a separate request mechanism for spot markets.

Do you have any good news about Spot order?

**Comment by baunov** [comment]
@pranaypratyush, having exactly same issue now. Any updates on it?
Private Spot endpoints (tried only GET ones) return

```
{
  ret_code: -1022,
  ret_msg: 'Signature for this request is not valid.',
  ext_code: null,
  ext_info: null,
  result: null
}
```

Package version: 2.1.8

**Comment by pranaypratyush** [comment]
> @pranaypratyush, having exactly same issue now. Any updates on it?

Not really. Library needs some work. Either I can fix or can someone else?

**Comment by tiagosiebler** [comment]
The spot client needs some work as signed requests with parameters are not working, as you have seen. There's some ongoing work in the `spotdev` branch but it's incomplete and subject to change. I do want to get this finished but haven't had the bandwidth:
https://github.com/tiagosiebler/bybit-api/pull/131

You can subscribe to the PR if you wish to stay informed.

**Comment by DavidKrpt** [comment]
+1, just encountered this issue. Ended up using these functions as a temporary solution.

Original post on SO - https://stackoverflow.com/a/72035877/10253105


```
import crypto from 'crypto'
import axios from 'axios';

const getSignature(parameters: any, secret: string) => {
  let orderedParams = "";

  Object.keys(parameters).sort().forEach(function(key) {
    orderedParams += key + "=" + parameters[key] + "&";
  });
  orderedParams = orderedParams.substring(0, orderedParams.length - 1);

  return crypto.createHmac('sha256', secret).update(orderedParams).digest('hex');
}

const postSpotOrder = async () => {
  const queryParams = {
  api_key: bybitApiKey,
  qty: 10,
  recvWindow: 10000,
  side: "BUY",
  symbol: "ETHUSDT",
  timestamp: Date.now(),
  type: "MARKET",
};

  const signature = getSignature(queryParams, bybitSecret);
  const postData = new URLSearchParams(queryParams);
  postData.append("sign", signature);

  let result = await axios(`https://api-testnet.bybit.com/spot/v1/order`, {
    withCredentials: true,
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
    },
    method: "POST",
    data: postData,
  });

  console.log(`Post Status`, result.status);
  console.log(`Post Body`, result.data);
};
```

**Comment by tiagosiebler** [comment]
Will be fixed in #150. Testers are welcome, it's currently published as beta before a full release in the coming few days: `npm i bybit-api@beta`
